dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -148,5 +156,36 @@ internal abstract class SdkFeature<T : Any, C : Configuration.Feature> {
         uploadScheduler.startScheduling()
     }
 
+    /**
+     * Since SDK v1.12.0, the Android SDK stores batch files in the cache directory instead
+     * of the files directory. This migration ensures we don't lose any important data when
+     * customers update their SDK.
+     */
+    protected fun migrateToCacheDir(
+        context: Context,
+        featureName: String,
+        internalLogger: Logger
+    ) {
+        val fileHandler = BatchFileHandler(internalLogger)
+        val config = FilePersistenceConfig()
+        val migrator = CacheFileMigrator(
+            fileHandler,
+            CoreFeature.persistenceExecutorService,
+            internalLogger
+        )
+        val filesDir = File(
+            context.filesDir,
+            FeatureFileOrchestrator.GRANTED_DIR.format(Locale.US, featureName)
```

Comment:
why do we move only granted files, and not pending?

Reply:
Because this migration happens at the start of a session, anything in the pending folder would be coming from a previous session, which, by definition, we don't have any consent for. Our policy in such a case is to just discard the data.